tasks/prime.py

```diff
@@ -362,6 +369,48 @@ def create_mto_shipment(self, overrides=None):
             self.add_stored(MTO_SHIPMENT, mto_shipment)
             return mto_shipment
 
+    @tag(MTO_SHIPMENT, "createMTOShipment", "expectedFailure")
+    @task
+    def create_mto_shipment_with_duplicate_agents(self, overrides=None):
+        # If moveTaskOrderID was provided, get that specific one. Else get any stored one.
+        object_id = overrides.get("moveTaskOrderID") if overrides else None
+
+        move_task_order = self.get_stored(MOVE_TASK_ORDER, object_id)
+        if not move_task_order:
+            logger.debug("createMTOShipment: ⚠️ No move_task_order found")
+            return (
+                None  # we can't do anything else without a default value, and no pre-made MTOs satisfy our requirements
+            )
+
+        agent_type = random.sample(["RELEASING_AGENT", "RECEIVING_AGENT"], 1).pop()
+        agent_override = {"id": ZERO_UUID, "mtoShipmentID": ZERO_UUID, "agentType": agent_type}
+        overrides_local = {
+            # Override moveTaskorderID because we don't want a random one
+            "moveTaskOrderID": move_task_order["id"],
+            # Set agents UUIDs to ZERO_UUID because we can't actually set the UUID on creation and guarantee two agents
+            "agents": [agent_override, agent_override],
+            # Set pickupAddress to ZERO_UUID because we can't actually set the UUID on creation
+            "pickupAddress": {"id": ZERO_UUID},
+            # Set destinationAddress to ZERO_UUID because we can't actually set the UUID on creation
+            "destinationAddress": {"id": ZERO_UUID},
+            # Set mtoServiceItems to empty to let the createMTOServiceItems endpoint do the creation
+            "mtoServiceItems": [],
+        }
+        # Merge local overrides with passed-in overrides
+        if overrides:
+            overrides_local.update(overrides)
+        payload = self.fake_request("/mto-shipments", "post", PRIME_API_KEY, overrides=overrides_local)
+
+        headers = {"content-type": "application/json"}
+        resp = self.client.post(
+            prime_path("/mto-shipments"),
+            name=prime_path("/mto-shipments — expected failure"),
+            data=json.dumps(payload),
+            headers=headers,
+            **self.user.cert_kwargs,
+        )
+        check_response(resp, "createMTOShipment", payload)
```

Comment:
As mentioned in the PR. Do we need this? We are generally using this throughout the tests but this may not be that useful for intentionally failed tests.

Reply:
Should we add a `check_failed_response` that does similar logging but expects it to not be a 2XX code? Asking because it looks like `check_responses` is defined in `tasks/base.py`

Reply:
Just added a change to modify `check_responses` to take an optional parameter of `expected_status_code` (defaults to 2xx) where it will do more verbose logging when the expected status code doesn't match the actual status code.